src/opnsense/mvc/app/controllers/OPNsense/Trust/Api/TrustBase.php

```diff
@@ -0,0 +1,108 @@
+     * keylen dropdown
+     * @var array
+     */
+    protected $keylens = [
+        "512" => ["value" => "512", "selected" => "0"],
+        "1024" => ["value" => "1024", "selected" => "0"],
+        "2048" => ["value" => "2048", "selected" => "1"],
+        "4096" => ["value" => "4096", "selected" => "0"],
+        "8192" => ["value" => "8192", "selected" => "0"]
+    ];
+
+    /**
+     * digest dropdown
+     * @var array
+     */
+    protected $digest_algs = [
+        "sha1" => ["value" => "SHA1", "selected" => "0"],
+        "sha224" => ["value" => "SHA224", "selected" => "0"],
+        "sha256" => ["value" => "SHA256", "selected" => "1"],
+        "sha384" => ["value" => "SHA384", "selected" => "0"],
+        "sha512" => ["value" => "SHA512", "selected" => "0"]
+    ];
+
+    /**
+     * Validation input strings
+     * @param $isset_fields
+     * @param $invalid_fields
+     * @param $post
+     * @param $method
+     * @return array
+     */
+    protected function Validation($isset_fields, $invalid_fields, $post, $method)
+    {
+        // Validation
+        $result = ["result" => "failed", "validations" => []];
+        foreach ($isset_fields as $field) {
+            /* check for bad control characters */
+            if (is_string($post[$field]) && preg_match("/[\\x00-\\x08\\x0b\\x0c\\x0e-\\x1f]/", $post[$field])) {
+                $result["validations"]["{$method}.{$field}"] = gettext("The field contains invalid characters.");
+            }
+            if (!isset($post[$field]) || empty($post[$field])) {
+                $result["validations"]["{$method}.{$field}"] = gettext("The field is required.");
+            }
+        }
+        foreach ($invalid_fields as $field) {
+            if (preg_match("/[\!\@\#\$\%\^\(\)\~\?\>\<\&\/\\\,\.\"\']/", $post[$field])) {
+                $result["validations"]["{$method}.{$field}"] = gettext("The field contains invalid characters.");
+            }
+        }
+        if (isset($post["dn_email"]) && preg_match("/[\!\#\$\%\^\(\)\~\?\>\<\&\/\\\,\"\']/", $post["dn_email"])) {
+            $result["validations"]["{$method}.dn_email"] = gettext("The field 'Distinguished name Email Address' contains invalid characters.");
+        }
+        if (isset($post["dn_commonname"]) && preg_match("/[\!\@\#\$\%\^\(\)\~\?\>\<\&\/\\\,\"\']/", $post["dn_commonname"])) {
+            $result["validations"]["{$method}.dn_commonname"] = gettext("The field 'Distinguished name Common Name' contains invalid characters.");
+        }
+        if (isset($post["keylen"]) && !in_array($post["keylen"], array_keys($this->keylens))) {
+            $result["validations"]["{$method}.keylen"] = gettext("Please select a valid Key Length.");
+        }
+        if (isset($post["digest_alg"]) && !in_array($post["digest_alg"], array_keys($this->digest_algs))) {
+            $result["validations"]["{$method}.digest_alg"] = gettext("Please select a valid Digest Algorithm.");
+        }
+        return $result;
```

Comment:
you should override performValidatoion like in #2018.

Reply:
Usually the input form matches a data model. So it is convenient to validate the data entered at the model level. In the case of a Trust/Certificates, we introduce some data, and on the basis of their calculated values to be written to the model. Therefore, you need to validate the entered data, because the data for the model can simply not now contain.